dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/AbstractCommandExecutor.java

```diff
@@ -125,6 +128,32 @@ public TaskResponse run(IShellInterceptorBuilder iShellInterceptorBuilder,
             result.setExitStatusCode(EXIT_CODE_KILL);
             return result;
         }
+        // the task instance needs fault tolerance
+        if(Objects.nonNull(taskRequest.getProcessId()) || StringUtils.isNotEmpty(taskRequest.getAppIds())){
+
+            int pid = taskRequest.getProcessId();
+            // Yarn task is determined by parsing whether the task log contains the content of the application
+            String applicationId  = String.join(TaskConstants.COMMA, LogUtils.getAppIds(taskRequest.getLogPath(), taskRequest.getAppInfoPath(),
+                    PropertyUtils.getString(APPID_COLLECT, DEFAULT_COLLECT_WAY)));
+            boolean isRunningTaskFaultTolerance = false;
+            if(applicationId.isEmpty()){
+                // not a Yarn task
+                isRunningTaskFaultTolerance = isProcessRunning(pid);
+            }else {
+                // is a Yarn task
+                isRunningTaskFaultTolerance = isApplicationRunning(applicationId);
+            }
+            //determines whether the task is running
+            if(Boolean.TRUE.equals(isRunningTaskFaultTolerance)){
+                logger.warn(
+                        "Can not recover tolerance fault task instance: {} , the task may be running and does not need to be repeated",
+                        taskInstanceId);
+                result.setExitStatusCode(EXIT_CODE_KILL);
+                cancelApplication();
+                return result;
```

Comment:
This kind of logic should in cancelApplication.

Reply:
Okay, I'll adjust